experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -0,0 +1,252 @@
+"""
+    get_coupler_config()
+
+Reads in the configuration file and job ID from the command line.
+A dictionary is constructed from the input configuration file, and this
+is returned along with the job ID.
+
+# Returns
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+- `job_id`: A unique identifier for the job
+"""
+function get_coupler_config()
+    ## coupler simulation default configuration
+    parsed_args = parse_commandline(argparse_settings())
+
+    ## modify parsed args for fast testing from REPL #hide
+    if isinteractive()
+        parsed_args["config_file"] =
+            isnothing(parsed_args["config_file"]) ? joinpath(pkg_dir, "config/ci_configs/interactive_debug.yml") :
+            parsed_args["config_file"]
+        parsed_args["job_id"] = "interactive_debug"
+    end
+
+    ## the unique job id should be passed in via the command line
+    job_id = parsed_args["job_id"]
+    @assert !isnothing(job_id) "job_id must be passed in via the command line"
+
+    ## read in config dictionary from file, overriding the coupler defaults in `parsed_args`
+    config_dict = YAML.load_file(parsed_args["config_file"])
+    config_dict = merge(parsed_args, config_dict)
+
+    return config_dict, job_id
+end
+
+"""
+    get_coupler_args!(config_dict)
+
+Extract the necessary arguments from the coupled configuration dictionary.
+This function may modify the input dictionary to remove unnecessary keys.
+
+# Arguments
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+
+# Returns
+- All arguments needed for the coupled simulation
+"""
+function get_coupler_args!(config_dict::Dict)
+    # Simulation-identifying information; Print `config_dict` if requested
+    config_dict["print_config_dict"] && @info(config_dict)
+    mode_name = config_dict["mode_name"]
```

Comment:
Here's a useful pattern for keyword dispatch.

```julia

"Type describing what to do with coupling"
abstract type AbstractCouplingMode end

"XYZ"
struct AMIP <: AbstractCouplingMode end

"XYZ"
struct Slabplanet <: AbstractCouplingMode end

# In the function 
allowed_modes = Dict(lowercase(string(nameof(t))) => t for t in subtypes(AbstractCouplingMode))
mode_name = lowercase(config_dict["mode_name"])

if haskey(allowed_modes, mode_name)
  coupling_mode = allowed_modes[mode_name]() 
else
  error("$mode_name is not an allowed coupling mode. Allowed modes are: $(keys(allowed_modes))")
end

# Then, you dispatch instead of doing `if-else`
```

The advantages of this pattern:
- Supports lower/upper cases
- Provides a useful error message
- Automatically extends to new subtypes, including subtypes that a user might define
- Allows us and users to dispatch
- Automatically catches invalid modes

Note, however, that these are probably different experiments, not really `CouplingModes`, so the concept is probably ill-posed to begin with.

Reply:
I like this idea. Since we're planning to separate the driver based on simulation type soon, I don't want to invest too much time in setting this up. But as we start working on that, we could use this structure